rustls/src/common_state.rs

```diff
@@ -845,6 +855,30 @@ pub(crate) trait State<Data>: Send + Sync {
     fn into_owned(self: Box<Self>) -> Box<dyn State<Data> + 'static>;
 }
 
+/// This is an output of [`State::handle()`]
+///
+/// It contains the next step in the state machine.
```

Comment:
nit: seems like this comment should also indicate it may contain the reason the state is blocked on the current state as well as mentioning it contains the next state.